mtz-wizard.html

```diff
@@ -1,34 +1,170 @@
+    <iron-pages
+      id="selector"
+      attr-for-selected="[[attrForSelected]]"
+      selectable="[[selectable]]"
+      selected-attribute="[[selectedAttribute]]"
+      selected-class="[[selectedClass]]"
+      selected-item="{{selectedItem}}"
+      selected="{{selected}}"
+      fallback-selection="[[fallbackSelection]]"
+      items="{{steps}}">
+      <template
+        is="dom-if"
+        if="[[active]]"
+        restamp>
+        <slot></slot>
+      </template>
+    </iron-pages>
   </template>
 
   <script>
     Polymer({
-
       is: 'mtz-wizard',
-
       properties: {
-        prop1: {
+        /* Determines if this wizard is active or not, used to attach/detach the steps from the DOM for data reset */
+        active: {
+          type: Boolean,
+          value: false
+        },
+        /* The attribute to use to match the selection key against */
+        attrForSelected: {
+          type: String,
+          value: 'name'
+        },
+        /* The fallback selection when nothing is selected */
+        fallbackSelection: {
+          type: String,
+          computed: '__computeFallbackSelection(steps, attrForSelected)'
+        },
+        /* This is a CSS selector string. If this is set, only items that match the CSS selector are selectable. */
+        selectable: {
+          type: String,
+          value: '[wizard-step],.wizard-step'
+        },
+        /* The currently selected step, null if no step is selected */
+        selected: {
           type: String,
-          value: 'mtz-wizard',
+          value: null,
+          notify: true
         },
+        /* The attribute to set on elements when selected */
+        selectedAttribute: {
+          type: String,
+          value: 'active'
+        },
+        /* The class to set on elements when selected. */
+        selectedClass: String,
+        /* The currently selected item */
+        selectedItem: {
+          type: Object,
+          notify: true,
+        },
+        /* The Set of HTMLElements that make up the steps */
+        steps: {
+          type: Array,
+          notify: true
+        }
       },
-
+      /**
+       * Fetches the model from all steps that have a form with a dirty model, returns as a map of step name to model.
+       *
+       * @param {HTMLElement[]} [steps = this.steps]
+       * @param {String} [attr = this.attrForSelected]
+       * @return {Object}
+       */
+      getFormData(steps = this.steps, attr = this.attrForSelected) {
+        const data = (steps || []).reduce((model, step) => {
+          if (step.form && step.dirty)
+            model[step.getAttribute(attr)] = step.form.serializeForm();
+          return model;
+        }, {});
+        return data;
+      },
+      /**
+       * Changes the currently active step to the next step in the array of steps, loops from end to beginning.
+       *
+       * @return {HTMLElement}
+       */
+      next() {
+        this.$.selector.selectNext();
```

Comment:
I find it odd that this and previous loop. Probably don't want this for a wizard.